payment/vote.go

```diff
@@ -214,33 +224,62 @@ func (service *Service) Vote(
 		return fmt.Errorf("error generating credential redemptions: %w", err)
 	}
 
-	// get a new VoteEvent to emit to kafka based on our input vote
-	voteEvent, err := NewVoteEvent(vote)
-	if err != nil {
-		return fmt.Errorf("failed to convert vote to kafka vote event: %w", err)
-	}
+	var credsByIssuer = map[string][]cbr.CredentialRedemption{}
 
-	// encode the event for processing
-	voteEventBinary, err := voteEvent.CodecEncode(service.codecs["vote"])
-	if err != nil {
-		return fmt.Errorf("failed to encode avro codec: %w", err)
+	if len(requestCredentials) > 0 {
+		for _, rc := range requestCredentials {
+			credsByIssuer[rc.Issuer] = append(credsByIssuer[rc.Issuer], rc)
+		}
 	}
 
-	rcSerial, err := json.Marshal(requestCredentials)
-	if err != nil {
-		return fmt.Errorf("failed to encode request credentials for vote drain: %w", err)
-	}
+	// for each issuer we will create a vote with a particular vote tally
+	for k, v := range credsByIssuer {
+		vote.VoteTally = int64(len(v))
+		// k holds the issuer name string, which has encoded in the funding source
+		// draw out the funding source and set it here.
 
-	// insert serialized event into db
-	if err = service.datastore.InsertVote(
-		ctx, VoteRecord{
-			RequestCredentials: string(rcSerial),
-			VoteText:           voteText,
-			VoteEventBinary:    voteEventBinary,
-		}); err != nil {
-		return fmt.Errorf("datastore failure vote_drain: %w", err)
-	}
+		_, sku, err := decodeIssuerID(k)
+		if err != nil {
+			return fmt.Errorf("failed to decode issuer name for sku: %w", err)
+		}
+
+		// get the part after the issuerSepartor
+		switch sku {
+		case UserWalletVoteSKU, AnonCardVoteSKU:
+			vote.FundingSource = sku
```

Comment:
Instead of the funding source being the SKU I think we would want the funding source to be like "anonymous-card" or "user-wallet". Thoughts?

Reply:
updated: https://github.com/brave-intl/bat-go/pull/303/commits/f626e23db8694674da001e6c1bbd4867c3606620